upgrade-tests/src/test/java/io/zeebe/test/ContainerStateRule.java

```diff
@@ -99,11 +100,24 @@ private void log(final String type, final String log) {
    *     false
    */
   public boolean hasElementInState(final String elementId, final String intent) {
+    return findLogContaining(
+        String.format("\"elementId\":\"%s\"", elementId),
+        String.format("\"intent\":\"%s\"", intent));
+  }
+
+  /** @return true if the message was found in the specified intent. Otherwise, returns false */
+  public boolean hasMessageInState(final String name, final String intent) {
+    return findLogContaining(
+        String.format("\"name\":\"%s\"", name), String.format("\"intent\":\"%s\"", intent));
+  }
+
+  // returns true if it finds a line that contains every piece.
+  boolean findLogContaining(final String... piece) {
     final String[] lines = broker.getLogs().split("\n");
 
     for (int i = lines.length - 1; i >= 0; --i) {
-      if (lines[i].contains(String.format("\"elementId\":\"%s\"", elementId))
-          && lines[i].contains(String.format("\"intent\":\"%s\"", intent))) {
+      final int finalI = i;
+      if (Arrays.stream(piece).allMatch(p -> lines[finalI].contains(p))) {
```

Comment:
We could compact the loop with the check a bit. For example: 
```
Arrays.stream(lines).anyMatch(line -> Arrays.stream(pieces).allMatch(line::contains))
```

Reply:
I considered that but I thought it was harder to read. I guess it's a matter of preference, I can change it